server/routes/associatedPrisoner/associatePrisonerPage.ts

```diff
@@ -0,0 +1,237 @@
+} from './associatePrisonerUtils'
+import PrisonerSearchService from '../../services/prisonerSearchService'
+
+export enum PageRequestType {
+  EDIT,
+  EDIT_SUBMITTED,
+}
+
+class PageOptions {
+  constructor(private readonly pageType: PageRequestType) {}
+
+  isPreviouslySubmitted(): boolean {
+    return this.pageType === PageRequestType.EDIT_SUBMITTED
+  }
+}
+
+export default class AssociatePrisonerPage {
+  pageOptions: PageOptions
+
+  constructor(
+    pageType: PageRequestType,
+    private readonly placeOnReportService: PlaceOnReportService,
+    private readonly prisonerSearchService: PrisonerSearchService
+  ) {
+    this.pageOptions = new PageOptions(pageType)
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const draftId = getDraftIdFromString(req.params.adjudicationNumber)
+    const { roleCode } = req.params
+    const fromDelete = req.query.personDeleted as string
+    const fromSearch = JSON.stringify(req.query.selectedPerson)?.replace(/"/g, '')
+    let roleAssociatedPrisoner = await this.readFromApi(draftId, user as User)
+
+    if (fromSearch) {
+      roleAssociatedPrisoner = {
+        prisonerNumber: roleAssociatedPrisoner.prisonerNumber,
+        currentAssociatedPrisonerNumber: fromSearch,
+      }
+    } else if (fromDelete) {
+      roleAssociatedPrisoner = {
+        prisonerNumber: roleAssociatedPrisoner.prisonerNumber,
+      }
+    }
+    const pageData = await this.getDisplayData(roleAssociatedPrisoner, null, user as User)
+
+    if (fromDelete) {
+      pageData.location = AssociatedPrisonerLocation.INTERNAL
+    }
+
+    renderData(res, draftId, roleCode, pageData, null)
+
+    delete req.session.redirectUrl
+  }
+
+  submit = async (req: Request, res: Response): Promise<void> => {
+    if (req.body.deleteUser) {
+      return this.delete(req, res)
+    }
+    if (req.body.searchUser) {
+      return this.search(req, res)
+    }
+    return this.save(req, res)
+  }
+
+  save = async (req: Request, res: Response) => {
+    const draftId = getDraftIdFromString(req.params.adjudicationNumber)
+    const { roleCode } = req.params
+    const { user } = res.locals
+    const { selectedAnswerId } = req.body
+
+    const { prisonerNumber } = await this.readFromApi(draftId, user as User)
+
+    const associatedPrisonersName = getAssociatedPrisonersName(req)
+    const associatedPrisonersNumber = getAssociatedPrisonersNumber(req, selectedAnswerId)
+
+    const roleAssociatedPrisoner = {
+      prisonerNumber,
+      currentAssociatedPrisonerName: associatedPrisonersName,
+      currentAssociatedPrisonerNumber: associatedPrisonersNumber,
+    }
+
+    const validationError = validateForm({
+      // eslint-disable-next-line no-nested-ternary
+      location: selectedAnswerId
+        ? selectedAnswerId === 'internal'
+          ? AssociatedPrisonerLocation.INTERNAL
+          : AssociatedPrisonerLocation.EXTERNAL
+        : AssociatedPrisonerLocation.UNKNOWN,
```

Comment:
Could we use a switch statement here or something else so we don't have to use a nested ternary?